httpx/http_service.go

```diff
@@ -126,7 +149,8 @@ func (t *RetryTransport) RoundTrip(ctx context.Context, req *http.Request) (*htt
 		if resp.StatusCode >= 500 {
 			return nil, &RetryableHTTPError{Path: req.URL.String(), StatusCode: resp.StatusCode}
```

Comment:
Oh, I misread your previous comment. Yeah, I think the behavior we'd want here is to return the last response and a nil error (unless of course there was an actual error from the underlying transport).